solana/programs/axelar-solana-its/src/event.rs

```diff
@@ -0,0 +1,126 @@
+#![allow(missing_docs)]
+use event_utils::*;
+use solana_program::pubkey::Pubkey;
+
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Event)]
+pub struct InterchainTransfer {
+    pub token_id: [u8; 32],
+    pub source_address: Pubkey,
+    pub destination_chain: String,
+    pub destination_address: Vec<u8>,
+    pub amount: u64,
+    pub data_hash: [u8; 32],
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Event)]
+pub struct InterchainTransferReceived {
+    pub command_id: [u8; 32],
+    pub token_id: [u8; 32],
+    pub source_chain: String,
+    pub source_address: Vec<u8>,
+    pub destination_address: Pubkey,
+    pub amount: u64,
+    pub data_hash: [u8; 32],
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Event)]
+pub struct TokenMetadataRegistered {
+    pub token_address: Pubkey,
+    pub decimals: u8,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Event)]
+pub struct LinkTokenStarted {
+    pub token_id: [u8; 32],
+    pub destination_chain: String,
+    pub source_token_address: Pubkey,
+    pub destination_token_address: Vec<u8>,
+    pub token_manager_type: u8,
+    pub params: Vec<u8>,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Event)]
+pub struct InterchainTokenDeploymentStarted {
+    pub token_id: [u8; 32],
+    pub token_name: String,
+    pub token_symbol: String,
+    pub token_decimals: u8,
+    pub minter: Vec<u8>,
+    pub destination_chain: String,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Event)]
+pub struct TokenManagerDeployed {
+    pub token_id: [u8; 32],
+    pub token_manager: Pubkey,
+    pub token_manager_type: u8,
+    pub params: Vec<u8>,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Event)]
+pub struct InterchainTokenDeployed {
+    pub token_id: [u8; 32],
+    pub token_address: Pubkey,
+    pub minter: Pubkey,
+    pub name: String,
+    pub symbol: String,
+    pub decimals: u8,
+}
```

Comment:
Same approach (using the macros) could be done for [governance ones](https://github.com/eigerco/solana-axelar/blob/main/solana/programs/axelar-solana-governance/src/events.rs) i guess - they need to be compliant with[ their js original implementation](https://github.com/axelarnetwork/axelar-gmp-sdk-solidity/blob/main/contracts/interfaces/IAxelarServiceGovernance.sol#L16). We are also emitting event there. Do you consider creating a new issue for doing it would make sense ?

Reply:
Oh, I didn't know Governance also has events. I think it's a good idea to create an issue to change and use the macro. The way the events are encoded there is slightly different (I followed the approach of the gateway and gas service), where each field is just logged as bytes. If we change the Governance, then Governance can also take advantage of part of the JS code I'm implementing (if needed).

Reply:
I created an [issue](https://github.com/eigerco/solana-axelar/issues/70) for governance, gateway and gasservice to be updated

Reply:
Awesome thanks ❤️  should they be in P6 (before mainet ?) ?